src/react-count-to.js

```diff
@@ -13,48 +12,59 @@ const CountTo = React.createClass({
     className: React.PropTypes.string,
   },
 
+  getDefaultProps() {
+    return {
+      from: 0,
+      delay: 100,
+      digits: 0,
+      onComplete: () => {},
+    };
+  },
+
   getInitialState() {
     return {
-      counter: this.props.from || 0,
+      counter: this.props.from,
     };
   },
 
   componentDidMount() {
-    this.start(this.props);
+    this.start();
   },
 
   componentWillReceiveProps(nextProps) {
-    if (!shallowequal(this.props, nextProps)) {
-      this.start(nextProps);
+    const props = this.props;
+
+    // no need to restart animation if onComplete / digits / className changes
+    if (nextProps.to !== props.to || nextProps.from !== props.from
```

Comment:
I really like this change and the fact that it makes the `shallowequal` dependency unnecessary.
However, I believe that at this point we could restart the counter only when `from` or `to` props change.

```javascript
componentWillReceiveProps(nextProps) {
  if (this.props.from !== nextProps.from || this.props.to !== nextProps.to) {
    this.start();
  }
},
```